src/MapNode.cuh

```diff
@@ -23,10 +24,10 @@ public:
      * Creates a `MapNode` object
      *
      * @param polyhedron Pointer to the polyhedron to create node on
-     * @param polyhedron_face_id The polyhedron's face to create node on
+     * @param polyhedron_face The polyhedron's face to create node on
      * @param coordinates Coordinates of node to create node at
      */
-    __device__ MapNode(Polyhedron *polyhedron, int polyhedron_face_id, SpacePoint coordinates);
+    __device__ MapNode(Polyhedron *const polyhedron, Face *polyhedron_face, SpacePoint coordinates);
```

Comment:
Why did you decided to add `const` to `polyhedron`?
`MapNode::polyhedron` field has `Polyhedron *` type
Also `Polyhedron *` type is used in all places where polyhedron is used

Reply:
> `MapNode::polyhedron` field has `Polyhedron *` type

Here the `const` qualifier, of course, only protects the argument from modifications only inside the function's body (because it's `T *const`, not `const T *`), so there is no matter what type do we assign it to (I mean `(Polyhedron *)a = (Polyhedron *const)b` is completely ok while the reversed is incorrect).

> Also `Polyhedron *` type is used in all places where polyhedron is used

Yes, I totally agree, this spoils the style. Don't remember why I've written it, so I'll remove it